### PR TITLE
Repo functions

### PR DESCRIPTION
**Pull request by bpinto**
Extracting common `git` calls into reusable functions.

### CODE REVIEW

File: pkg/omf/cli/omf.repo.clone.fish [hunk 1]
```diff
@@ -0,0 +1,3 @@
+function omf.repo.clone.fish -a url path
```
Comment (derekstavis) [inline pkg/omf/cli/omf.repo.clone.fish]: Oops @ `omf.repo.clone.fish`

Comment (bpinto) [inline pkg/omf/cli/omf.repo.clone.fish]: Sorry?

Comment (derekstavis) [inline pkg/omf/cli/omf.repo.clone.fish]: The function name is prepended by `.fish` :dancers:

Comment (bpinto) [inline pkg/omf/cli/omf.repo.clone.fish]: :dancers:

File: pkg/omf/cli/omf.repo.pull.fish [hunk 2]
```diff
@@ -0,0 +1,41 @@
+function omf.repo.pull
+  if test (git config --get remote.upstream.url)
```
Comment (derekstavis) [inline pkg/omf/cli/omf.repo.pull.fish]: I'm not sure, isn't `command git` needed in this case?

Comment (bobthecow) [inline pkg/omf/cli/omf.repo.pull.fish]: :+1:

Comment (bpinto) [inline pkg/omf/cli/omf.repo.pull.fish]: Thanks for pointing it out!

Comment (bpinto) [inline pkg/omf/cli/omf.repo.pull.fish]: Fixed.

**Comment by bpinto** [comment]
@derekstavis @bobthecow fixes applied.

**Comment by bobthecow** [comment]
lgtm :+1:

**Comment by derekstavis** [comment]
I couldn't find a sane way to test. How you guys tested it?

**Comment by bpinto** [comment]
I have installed a plugin via URL: `omf install https://github.com/oh-my-fish/plugin-osx` and also I have changed my local master to be outdated from `origin/master` so running `omf update` on another branch caused my `master` branch to be updated.

**Comment by derekstavis** [comment]
This is my output when I `omf update`:

```
 ~/.l/s/omf_bak  ➦ 6238cd9  ±  omf update
Updating Oh My Fish...
fatal: ref HEAD is not a symbolic ref
Oh My Fish failed to update.
Please open a new issue here → github.com/oh-my-fish/oh-my-fish/issues
Updating agnoster theme...
✔ agnoster theme up to date.
Updating foreign-env package...
✔ foreign-env package up to date.
Error: migrate package already installed.
Updating osx package...
✔ osx package up to date.
Trying to clone from URL...
fatal: repository 'pkg-mvn' does not exist
Could not install package pkg-mvn.
Updating tiny package...
✔ tiny package up to date.
Updating title package...
✔ title package up to date.
```

**Comment by bpinto** [comment]
Can you investigate it? It seems like it could be a problem with how we fetch the repo (grafted).

**Comment by derekstavis** [comment]
The function `omf.repo.pull` was missing a `return 0`. I've pushed the fix to this PR.

**Comment by derekstavis** [comment]
I think this is good to merge, so I'm doing it.

**Pull request merged**